src/test/unit/algorithms/ConnectionsPerformanceTest.cpp

```diff
@@ -280,9 +251,46 @@ vector<CellIdx> ConnectionsPerformanceTest::computeSPWinnerCells(
   return vector<CellIdx>(winnerCells.begin(), winnerCells.end());
 }
 
-} // end namespace nupic
 
-int main(int argc, char *argv[]) {
-  ConnectionsPerformanceTest test = ConnectionsPerformanceTest();
-  test.RunTests();
+
+// TESTS
+ConnectionsPerformanceTest t;
+
+void SetUp() {
+  srand(SEED);
+  t = testing::ConnectionsPerformanceTest();
+}
+
+/**
+ * Tests typical usage of Connections with Temporal Memory.
+ */
+TEST(ConnectionsPerformanceTest, testTM) {
+	auto tim = t.runTemporalMemoryTest(2048, 40, 20, 100, "temporal memory");
+	ASSERT_LE(tim, 1.5f); //there are times, we must be better. Bit underestimated for slow CI
```

Comment:
Idea is to keep an eye on the performance/speed by the CI. If someone severely breaks the performance, these tests should trigger! 

Problem is the CI is slow compared to normal PC, and that the load there varies -> we'll have to have a very large "buffer" for the slowest CI. 

If someone improves performance significantly, these numbers should be updated (lowered) to reflect that. 
Once set, we ideally never should increase the numbers (means losing performance).